modules/prebidServerBidAdapter/index.js

```diff
@@ -395,13 +397,97 @@ const OPEN_RTB_PROTOCOL = {
           return { w, h };
         });
 
-        banner = {format};
+        mediaTypes['banner'] = {format};
       }
 
-      let video;
       const videoParams = utils.deepAccess(adUnit, 'mediaTypes.video');
       if (!utils.isEmpty(videoParams)) {
-        video = videoParams;
+        mediaTypes['video'] = videoParams;
+      }
+
+      const nativeParams = processNativeAdUnitParams(utils.deepAccess(adUnit, 'mediaTypes.native'));
+      if (nativeParams) {
+        let assetCounter = 200;
+        try {
+          mediaTypes['native'] = {
+            request: JSON.stringify({
+              // TODO: determine if following OpenRTB recommended properties can be defined or are needed
+              // context: int,
+              // plcmttype: int,
+              // privacy: int
+              assets: Object.keys(nativeParams).reduce((assets, type, index) => {
+                let params = nativeParams[type];
+
+                function newAsset(obj) {
+                  return Object.assign({
+                    id: assetCounter++,
+                    required: params.required ? 1 : 0
+                  }, obj ? utils.cleanObj(obj) : {});
+                }
+
+                if (
+                  (type === 'image' || type === 'icon')
+                ) {
+                  let imgTypeId = ({
+                    'icon': 1,
+                    'image': 3
+                  })[type];
+
+                  if (Array.isArray(params.sizes)) {
+                    let sizes = params.sizes;
+                    if (!Array.isArray(sizes[0])) { // support single size or an array of sizes
+                      sizes = [sizes]
+                    }
+                    // TODO: Currently creating multiple image assets for each size.  Figure out if this is correct!
```

Comment:
Regarding request objects for native images, the OpenRTB Native 1.2 spec when implemented for an icon assets looks similar to the following:
```javascript
// image
{
  "id": 200,
  "required": 1,
  "img": {
    "type": 3,
    "wmin": 300,
    "hmin": 250
  }
}

// icon
{
  "id": 201,
  "required": 1,
  "img": {
    "type": 1,
    "w": 50,
    "h": 50
  }
}
```
Currently image and icon assets are defined on the prebid adUnit [as defined here](http://prebid.org/dev-docs/adunit-reference.html#adunitmediatypesnative) in a way specifying multiple sizes and/or aspect ratios.  I have coded this current implementation to loop over the sizes and create separate image assets for each size, however this is _probably_ not correct with the required for each as, in prebid.js, each size means the adUnit could be any one of those sizes and not require all sizes.  As I see it, to solve this problem the request could be mapped into separate image assets (as I have done here in the code) but remove the required flag, or a single image asset could be sent and the `ext` object could be somehow utilized to send multiple sizes.

Example:
```javascript
pbjs.addAdunits([{
	code: '/19968336/prebid_native_example_1',
	mediaTypes: {
		'native': {
			'image': {
        'required': true,
        'sizes': [[300, 250], [300, 600]]
      }
		}
	}
}]);

// gets represented in OpenRTB Native request as multiple assets
{
  "assets": [
    {
      "id": 200,
      "img": {  // notice lack of required since both can't be required
        "type": 3,
        "w": 300,
        "h": 250
      }
    }, {
      "id": 201,
      "img": {
        "type": 3,
        "w": 300,
        "h": 600
      }
    }
  ]
}

// or ext is used
{
  "assets": [
    {
      "id": 200,
      "required": 1,
      "img": {
        "type": 3,
        "ext": {
          "sizes": [{
            "w": 300,
            "h": 250
          }, {
            "w": 300,
            "h": 600
          }]
        }
      }
    }
  ]
}

// or something else entirely...
```

For aspect ratios, currently OpenRTB only supports a `wmin` and `hmin` property, whereas the Prebid.js documentation lists only `min_width`, `ratio_height`, and `ratio_width` being the only valid properties on an array of objects.  I think the same could be done as above, mapping each aspect ratio object in the array to an image asset, however only `wmin` would be set and the ratios would be set as `ext.ratio_height` and `ext.ratio_width`.

I'm also assuming here that if both `sizes` and `aspect_ratios` are specified on the adUnit that they don't apply to the same asset (i.e. `sizes[0] and `aspect_ratios[0]` doesn't mean there is only one image asset that must meet both constraints)?